regression/barebones/barebones_suite_test.go

```diff
@@ -28,24 +27,42 @@ func getFabricTestDir() (string, error) {
 	return fabricTestPath, nil
 }
 
-func getTestStatusFromReportFile(filePath, action string) string {
+var (
+	fabricTestDir string
+	testDataDir   string
 
-	invokeStatus := "INVOKE Overall TEST RESULTS PASSED"
-	queryStatus := "QUERY Overall TEST RESULTS PASSED"
-	data, err := ioutil.ReadFile(filePath)
-	if err != nil {
-		return fmt.Sprintf("%s", err)
-	}
-	fileData := string(data)
-	switch action {
-	case "invoke":
-		if strings.Contains(fileData, invokeStatus) {
-			return invokeStatus
-		}
-	case "query":
-		if strings.Contains(fileData, queryStatus) {
-			return queryStatus
-		}
+	action          string
+	networkSpecPath string
+
+	kubeConfig    string
+	envKubeConfig bool
+	containerType string
+)
+
+var _ = BeforeSuite(func() {
+	// set up dir variables
+	fabricTestDir, _ = getFabricTestDir()
+	testDataDir = path.Join(fabricTestDir, "regression/testdata")
+
+	// set up input file variables
+	networkSpecPath = path.Join(testDataDir, "barebones-network-spec.yml")
+
+	// get kubeConfig from environment variable
+	kubeConfig, envKubeConfig = os.LookupEnv("KUBECONFIG")
+	if envKubeConfig {
+		containerType = "k8s"
+	} else {
+		containerType = "docker"
 	}
-	return "Overall TEST RESULTS FAILED"
-}
+
+	// bring up network
+	action = "up"
+	err := launcher.Launcher(action, containerType, kubeConfig, networkSpecPath)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	action = "down"
+	err := launcher.Launcher(action, containerType, kubeConfig, networkSpecPath)
```

Comment:
containerType and kubeConfig will be empty as AfterSuite is not having from line 51 to 56. May be make a common function for kubeConfig and containerType setup and then call it in BeforeSuite and AfterSuite.

Reply:
add a fun to get kube env, then call this func from both BeforeSuite and AfterSuite